src/main/java/org/codice/ddf/catalog/direct/CatalogMethods.java

```diff
@@ -0,0 +1,539 @@
+
+import static org.apache.commons.lang3.tuple.ImmutablePair.of;
+import static org.codice.jsonrpc.JsonRpc.INTERNAL_ERROR;
+import static org.codice.jsonrpc.JsonRpc.INVALID_PARAMS;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeRegistry;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.impl.PropertyNameImpl;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.SourceInfoRequestSources;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.io.Serializable;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.codice.jsonrpc.DocMethod;
+import org.codice.jsonrpc.Error;
+import org.codice.jsonrpc.JsonRpc;
+import org.codice.jsonrpc.MethodSet;
+import org.geotools.filter.SortByImpl;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.opengis.filter.Filter;
+import org.opengis.filter.sort.SortBy;
+import org.opengis.filter.sort.SortOrder;
+
+public class CatalogMethods implements MethodSet {
+
+  private static final String ISO_8601_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+  public static final String ATTRIBUTES = "attributes";
+
+  //  private final transient SimpleDateFormat dateFormat = new
+  // SimpleDateFormat(ISO_8601_DATE_FORMAT);
+
+  private final Map<String, DocMethod> METHODS;
+
+  {
+    Builder<String, DocMethod> builder = ImmutableMap.builder();
+    builder.put("ddf.catalog/create", new DocMethod(this::create, ""));
```

Comment:
```suggestion
    builder.put("ddf.catalog/create", new DocMethod(this::create, "Takes the specified parameters (metacards) and calls CatalogFramework::create.`params` takes: `metacards(Required, value: List(Object(`metacardType`:string, `attributes`:Object(Required, `id`: String))) "));
```

Reply:
I know we want examples in here, but for the interim here's at least some semi-instructive docstrings